geth/transactions/txqueue_manager_test.go

```diff
@@ -345,3 +355,28 @@ func (s *TxQueueTestSuite) TestLocalNonce() {
 	resultNonce, _ = s.manager.localNonce.Load(tx.Args.From)
 	s.Equal(uint64(nonce)+1, resultNonce.(uint64))
 }
+
+func (s *TxQueueTestSuite) TestContractCreation() {
+	key, _ := crypto.GenerateKey()
+	testaddr := crypto.PubkeyToAddress(key.PublicKey)
+	genesis := core.GenesisAlloc{
+		testaddr: {Balance: big.NewInt(100000000000)},
+	}
+	backend := backends.NewSimulatedBackend(genesis)
+	selectedAccount := &account.SelectedExtKey{
+		Address:    testaddr,
+		AccountKey: &keystore.Key{PrivateKey: key},
+	}
+	s.manager.ethTxClient = backend
+	tx := Create(context.Background(), SendTxArgs{
+		From:  testaddr,
+		Input: hexutil.Bytes(gethcommon.FromHex(contract.ENSBin)),
+	})
+	s.NoError(s.manager.QueueTransaction(tx))
+	hash, err := s.manager.CompleteTransaction(tx.ID, selectedAccount)
+	s.NoError(err)
+	backend.Commit()
+	receipt, err := backend.TransactionReceipt(context.TODO(), hash)
+	s.NoError(err)
+	s.Equal(crypto.CreateAddress(testaddr, 0), receipt.ContractAddress)
```

Comment:
This is because `NewSimulatedBackend` returns `ContractAddress` as a value of `From` field during contract creation?

Reply:
hm no, thats because transactions receipt has a ContractAddress field. it is not specific for simulated backend

Reply:
I don't see how `From` value can equal `ContractAddress` when creating a contract, hence was my question.